libsharedringbuffer/src/shared_ringbuffer.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2022, UNSW (ABN 57 195 873 179)
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <shared_ringbuffer/shared_ringbuffer.h>
+
+void ring_init(ring_handle_t *ring, ring_buffer_t *avail, ring_buffer_t *used, notify_fn notify, int buffer_init)
+{
+    ring->used_ring = used;
+    ring->avail_ring = avail;
+    ring->notify = notify;
```

Comment:
Do we consider passing `NULL` for any of these an error actually? We could put an assert here for development convenience, so this is caught and avoids potentially having to debug strange errors later.